vcd/resource_vcd_vapp_vm.go

```diff
@@ -617,6 +617,13 @@ func vmSchemaFunc(vmType typeOfVm) map[string]*schema.Schema {
 			Computed:    true, // As it can get populated automatically by VDC default policy
 			Description: "VM placement policy ID. Has to be assigned to Org VDC.",
 		},
+		"security_tags": {
```

Comment:
`read`  function of VM is shared with a data source `vcd_vapp_vm` so if you could also add the same field to `datasource_vcd_vapp_vm.go`. Terraform data source will crash otherwise by trying to set non-existing field.

In data source the field should be `computed` only:
```go
	"security_tags": {
			Type:        schema.TypeSet,
			Computed:    true,
			Description: "Security tags assigned to this VM",
			Elem:        &schema.Schema{Type: schema.TypeString},
		},
```